lib/bn/fqp.ex

```diff
@@ -0,0 +1,308 @@
+defmodule BN.FQP do
+  defstruct [:coef, :modulus_coef, :dim]
+
+  alias BN.FQ
+
+  @type t :: %__MODULE__{
+          coef: [FQ.t()],
+          modulus_coef: [integer()]
+        }
+
+  @spec new([integer()], [integer()], keyword()) :: t()
+  def new(coef, modulus_coef, params \\ []) do
+    modulus = params[:modulus] || FQ.default_modulus()
```

Comment:
Could also use `Keyword.get(params, :modulus, FQ.default_modulus())`. I think what you have works just fine, so purely stylistic.